content/docs/latest/spire/understand/concepts.md

```diff
@@ -88,12 +88,26 @@ This bootstrap bundle is a default configuration, and should be replaced with cu
 10. The server performs node resolution, to verify additional properties about the agent node and update its registration entries accordingly. For example, if the node was attested using  Microsoft Azure Managed Service Identity (MSI). The resolver extracts the Tenant ID and Principal ID from the agent SPIFFE ID and uses the various Azure services to get information for building an additional set of selectors.
 11. The server issues an SVID to the agent, representing the identity of the agent itself.  
 12. The agent contacts the server (using its SVID as its TLS client certificate) to obtain the registration entries it is authorized for and to ask the server to sign workload SVIDs.
-13. Now fully bootstrapped, the agent turns on the Workload API.  
-14. A workload calls the Workload API to request an SVID.
-15. The agent initiates the workload attestation process by calling its workload attestors, providing them with the process ID of the workload process.  
-16. Attestors use the process ID to discover additional information about the workload
-17. The attestors return the discovered information to agent in the form of selectors
-18. The agent determines the workload's identity by comparing discovered selectors to registration entries, and returns the correct SVID to the workload when the workload asks for it.  
+13. The server sees the SPIFFE ID of the agent's own SVID (the one used as it's TLS client certificate), and fetches all [authorized registration entries](#authorized-registration-entries). All Those are the workload entries signed and return to the agent.
+14. Now fully bootstrapped, the agent turns on the Workload API.  
+15. A workload calls the Workload API to request an SVID.
+16. The agent initiates the workload attestation process by calling its workload attestors, providing them with the process ID of the workload process.  
+17. Attestors use the process ID to discover additional information about the workload
+18. The attestors return the discovered information to agent in the form of selectors
+19. The agent determines the workload's identity by comparing discovered selectors to registration entries, and returns the correct SVID (already in stock) to the workload when the workload asks for it.  
+
+## Authorized registration entries
+
+When the server sees the SPIFFE ID of the agent's own SVID, he fetches and signs all registration entries, for with the agent's SPIFFE ID is authorized. The selection query is construed as follows:
+
+1. Fetch all registration entries that have the agent's SPIFFE ID set as parentID
+2. Determin from the server's database what additional properties the specific agent is associated with.
+3. Fetch any SPIFFE ID that &mdash; at least &mdash; declares one selection on any of those additional properties. \*
+4. Also recursively fetch all decendants of the resulting list of SPIFFE IDs.
```

Comment:
```suggestion
4. Also recursively fetch all descendants of the resulting list of SPIFFE IDs.
```

Reply:
Recursiveness is not obvious but true. It happens over the childEntries pathway on recursive calls while on _non-node_ child entries, the `f.dataStore.GetNodeSelectors` would just not return a valid node selector.
https://github.com/spiffe/spire/blob/9b3b46948015b3412be79c4ef8b402555f6eb4eb/pkg/server/util/regentryutil/fetch.go#L67-L73

Reply:
>  I don't think either line 3 ... is true

What would be the matching logic of: https://github.com/spiffe/spire/blob/9b3b46948015b3412be79c4ef8b402555f6eb4eb/pkg/server/util/regentryutil/fetch.go#L138-L141 ?
I tried to understand `datastore.BySelectors_MATCH_SUBSET` but wasn't completely sure, so I complemented with my own observation. Previously I assumed that all the node's selectors must match, but that would krassly contradict how node pool aliases are construed, so I figured the interpretation had to be inverse: _at least one_.

Reply:
Oh, now I see what you are saying. This is a process that is internal to the server and not exposed on the workload API. You are correct about how it works. Do you think we should go into this level of detail?

Reply:
> Do you think we should go into this level of detail?

Yes, it cost me a day and a half to figure out and it's not explained anywhere else what parent_ids are for. There have been similar confusions: https://github.com/spiffe/spire/issues/1822 :wink: 

It is really critical to understand if you want to use spiffe in non trivial ways off side of the quickstart tutorials.

Reply:
Ok. What if we say "Query the database for any registration entries that have the agent's SPIFFE ID listed as their "parent SPIFFE ID", or recursively have any of THOSE entries listed as a parent ID, and so on."
Mostly want to stress that this is a database query happening entirely inside the SPIRE server.

Reply:
Agreed and addressed in c7fc9b7a0c1a760ed97e875f3686e417e57c6313 to make this more clear.